### ISSUE TITLE
Allow "filenames with spaces" in the DatArchive API

### ISSUE THREAD

**Issue by pmario** [beakerbrowser/beaker#577]
Operation System: ubuntu 17.04
Beaker Version: 0.7.3

## to reproduce 

 - manually create a file named: "a b.txt" content: test in the Site director
 - open it with beaker: `dat://<key>/a b.txt`
 - URL shows: `dat://<key>/a%20b.txt`
   - isn't nice but the file is reachable. FireFox does much better here.  

now try this: 

```
var archive = new DatArchive(datUrl)
await archive.writeFile('/b c.txt', 'world', {encoding: utf8})
```

I get:
 
```
err = InvalidPathError: Path contains invalid characters at createError (/home/mario/git/beaker/app/webview-preload.build.js:1555:16) at onCbReply (/home/mario/git/beaker/app/webview-preload.build.js:1538:17) at EventEmitter.onIPCMessage (/home/mario/git/beaker/app/webview-preload.build.js:1477:14) at emitMany (events.js:127:13) at EventEmitter.emit (events.js:204:7) {name: "InvalidPathError", invalidPath: true, stack: "InvalidPathError: Path contains invalid characters…27:13)↵ at EventEmitter.emit (events.js:204:7)", message: "Path contains invalid characters"}
401
        debugger
402
        return callback(err)})
```

## expected

 - it should be possible to create any filename, that the OS allows

**Comment by pfrazee** [comment]
Yeah this is something that I did because I wanted Dat filepaths to always be hygienic inside of URLs. If we support spaces, we end up with `%20` interspersed. That example would be `/b%20c.txt`

I recognize that might not be a good enough reason to disallow spaces. Happy to hear other opinions.

**Comment by pfrazee** [comment]
Oh yeah. As you pointed out, I cant really control this outside of the DatArchive API

Ok, I'll relax this constraint

**Comment by pmario** [comment]
>That example would be /b%20c.txt
>
>I recognize that might not be a good enough reason to disallow spaces. Happy to hear other opinions.

If you have a look at FireFox it displays the link nicely in the URL input area. .. If you copy paste the URL it inserts the url-encoding again. ... So the display is nice, but the URL is ugly :)

with tiddlywiki we have a function like this: `$tw.utils.transliterate(baseFilename.replace(/ |<|>|\:|\"|\||\?|\*|\^/g,"_"));` which replaces unwanted chars in filenames with underline.

I just added "space" again, which we removed some time ago, to fix my problem :)

This solution isn't perfect since eg: `$:/aa` and `$/:aa` will produce `$__aa`, which may overwrite content ... but we didn't get issues so far. 

---------

We also use transliterations for Cyrillic chars like so (list is shortened): 

```
/*
Transliterate string from eg. Cyrillic Russian to Latin
*/
var transliterationPairs = {
	"Ё":"YO",
	"Й":"I",
	"Ц":"TS",
	"У":"U",
	"К":"K",
	"Е":"E",
	"Н":"N",
	"Г":"G",
	"Ш":"SH",
	"Щ":"SCH",
```

All the quirks is needed, since users want to use every possible character for their content titles, which are partially reflected in the filename ... 

just some thoughts.
